### PR TITLE
Quiz-functionality

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added Quiz Functionality.
-Home page: Quizella has 5 Categories, user can search quiz through category.
-Rules page: It contains all the rules and instructions.
-Question Page: It contains 10 Questions of the selected category.
-Result page: It contains the total score and all the questions with correct and incorrect answers.
Screenshot (329)
Screenshot (330)
Screenshot (331)
Screenshot (332)

### CODE REVIEW

File: src/components/Modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            className="modal__btn--demo modal__btn--no"
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            No
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+};
```
Comment (munjalSaksham71) [inline src/components/Modal/Modal.jsx:27]: Liked that you have created your own modal

File: src/Pages/Rules-Page/QuizRule.jsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import "./quizrule.css";
 import { useNavigate } from "react-router-dom";
+import { useQuiz } from "../../Context/QuizContext";
 export const QuizRule = () => {
   const navigate = useNavigate();
+  const { state } = useQuiz();
```
Comment (munjalSaksham71) [inline src/Pages/Rules-Page/QuizRule.jsx:6]: this is fine - just a suggestion you can destructure ```currentQuiz``` like: ```{ state: {currentQuiz} }```

and there you have to add ```currentQuiz.category```

**Review by munjalSaksham71 — APPROVED** [review]
UI looks pretty good - Loved how you show results.

**Review by Manju-sree — APPROVED** [review]
code looks clean and understandable, maintained proper folder structure as well. code is formatted as well and UI looks clean and good. I think you can go ahead and merge this PR.

File: src/Pages/Rules-Page/QuizRule.jsx (hunk 2 above)
Comment (SaurabhChawda) [reply src/Pages/Rules-Page/QuizRule.jsx:6]: Okay, I will update in the next PR.
ThankYou

**Comment by SaurabhChawda** [comment]
> code looks clean and understandable, maintained proper folder structure as well. code is formatted as well and UI looks clean and good. I think you can go ahead and merge this PR.

ThankYou so much.

**Pull request merged**